src/java.base/share/classes/java/lang/VirtualThread.java

```diff
@@ -1018,15 +1038,20 @@ public String toString() {
         sb.append("]/");
         Thread carrier = carrierThread;
         if (carrier != null) {
-            // include the carrier thread state and name when mounted
-            synchronized (carrierThreadAccessLock()) {
-                carrier = carrierThread;
-                if (carrier != null) {
-                    String stateAsString = carrier.threadState().toString();
-                    sb.append(stateAsString.toLowerCase(Locale.ROOT));
-                    sb.append('@');
-                    sb.append(carrier.getName());
+            notifyJvmtiDisableSuspend(true);
+            try {
+                // include the carrier thread state and name when mounted
```

Comment:
This one too, can you move the comment to before the notifyJvmtiDisableSuspend.

Reply:
Moved both comments out of try blocks.
What about this one (it seems we would wont to do the same) ? :
```
                notifyJvmtiDisableSuspend(true);
                try {
                    // unpark carrier thread when pinned
                    synchronized (carrierThreadAccessLock()) {
                        Thread carrier = carrierThread;
                        if (carrier != null && ((s = state()) == PINNED || s == TIMED_PINNED)) {
                            U.unpark(carrier);
                        }
                    }
                } finally {
                    notifyJvmtiDisableSuspend(false);
                }
```

Reply:
Moved 3 comments out of try blocks.